index.js

```diff
@@ -113,18 +113,32 @@ class TimeoutError extends Error {
 
 const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
 
-const timeout = (promise, ms, abortController) => Promise.race([
-	promise,
-	(async () => {
-		await delay(ms);
-		if (abortController) {
-			// Throw TimeoutError first
-			setTimeout(() => abortController.abort(), 1);
-		}
+const timeout = (promise, ms, abortController) => {
+	let successful = false;
+	return Promise.race([
+		(async () => {
+			try {
+				return await promise;
+			} finally {
+				successful = true;
+			}
+		})(),
+		(async () => {
+			await delay(ms);
 
-		throw new TimeoutError();
-	})()
-]);
+			if (successful) {
+				return;
+			}
+
+			if (abortController) {
+				// Throw TimeoutError first
+				setTimeout(() => abortController.abort(), 1);
+			}
+
+			throw new TimeoutError();
+		})()
+	]);
+};
```

Comment:
```js
// `Promise.race()` workaround (#91)
const timeout = (promise, ms, abortController) => new Promise((resolve, reject) => {
	promise.then(resolve).catch(reject);
	delay(ms).then(() => {
		reject(new TimeoutError());
		abortController.abort();
	});
});
```

This does the same as `Promise.race()`, but the debugger just omits the error as expected :)

Reply:
Smart and elegant! I’m quite impressed!

I guess with your proposed solution there’s no need for my PR and you can just apply the change by yourself.